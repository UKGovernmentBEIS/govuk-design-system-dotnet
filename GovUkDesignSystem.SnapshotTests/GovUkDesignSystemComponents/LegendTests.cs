using System.Threading.Tasks;
using GovUkDesignSystem.GovUkDesignSystemComponents;
using GovUkDesignSystem.SnapshotTests.Helpers;
using Xunit;

namespace GovUkDesignSystem.SnapshotTests.GovUkDesignSystemComponents
{
    public class LegendTests : SnapshotTestBase
    {
        private LegendViewModel DefaultLegendViewModel()
        {
            return new LegendViewModel
            {
                Text = "text",
                Html = o => "html",
                Classes = "cssClass",
                IsPageHeading = false
            };
        }

        [Fact]
        public async Task Render_AllValues()
        {
            // Act & Assert
            await VerifyPartial("Legend", DefaultLegendViewModel());
        }

        [Fact]
        public async Task Render_NoHtml()
        {
            // Arrange
            var viewModel = DefaultLegendViewModel();
            viewModel.Html = null;

            // Act & Assert
            await VerifyPartial("Legend", viewModel);
        }

        [Fact]
        public async Task Render_AsPageHeading()
        {
            // Arrange
            var viewModel = DefaultLegendViewModel();
            viewModel.IsPageHeading = true;

            // Act & Assert
            await VerifyPartial("Legend", viewModel);
        }
    }
}
