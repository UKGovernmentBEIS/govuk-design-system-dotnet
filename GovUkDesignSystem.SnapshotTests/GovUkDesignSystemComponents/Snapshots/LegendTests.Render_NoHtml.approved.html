<legend class="govuk-fieldset__legend cssClass">

text            </legend>